sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWritesShardedRecords.java

```diff
@@ -647,6 +662,12 @@ public void process(
                             context.offset + context.protoRows.getSerializedRowsCount() - 1,
                             false)));
             flushesScheduled.inc(context.protoRows.getSerializedRowsCount());
+
+            if (successfulRowsTag != null) {
+              for (ByteString protoBytes : context.protoRows.getSerializedRowsList()) {
+                o.get(successfulRowsTag).output(appendClientInfo.get().toTableRow(protoBytes));
```

Comment:
can toTableRow() throw an exception here? A different conversion approach is used in another place (DynamicMessage.parseFrom(...)).

Reply:
toTableRow simply calls DynamicMessage.parseFrom